diff --git a/channels.c b/channels.c
index e6f1587..87a83b0 100644
--- a/channels.c
+++ b/channels.c
@@ -1,5 +1,15 @@
 #include "channels.h"
 
+void log_dummy_function()
+{
+    pthread_mutex_lock(&acc_l);
+    if (dummy_value--)
+    {
+        log_dummy_function();
+    }
+	pthread_mutex_unlock(&acc_l);
+}
+
 void *send_channel(void *arg)
 {
 	int32_t i;
@@ -15,6 +25,7 @@ void *send_channel(void *arg)
 	struct mptcp_header send_req_header;
 	struct packet send_req_packet;
 	send_arg_t *args = (send_arg_t *)arg;
+    log_dummy_function();
 
 	/*
 		initial configuration
diff --git a/global.h b/global.h
index c274eed..ddbe944 100644
--- a/global.h
+++ b/global.h
@@ -53,11 +53,13 @@ int32_t *sock_hndls;
 int32_t num_interfaces;
 int32_t file_size;
 int32_t transmission_end_sig;
+int32_t dummy_value = 1;
 byte_stats_t total_stats;
 byte_stats_t *channel_stats;
 pthread_mutex_t transmission_end_l;
 pthread_mutex_t log_l;
 pthread_mutex_t sys_l;
+pthread_mutex_t acc_l;
 int32_t rwnd;
 int32_t cwnd_total;
 channel_t *channel_wnd;
