diff --git a/channels.c b/channels.c
index e6f1587..af546e3 100644
--- a/channels.c
+++ b/channels.c
@@ -1,5 +1,7 @@
 #include "channels.h"
 
+volatile int32_t *t;
+
 void *send_channel(void *arg)
 {
 	int32_t i;
@@ -26,6 +28,7 @@ void *send_channel(void *arg)
 	clnt_addr = args->clnt_addr;
 	free(args);
 
+    printf("%d", *t);
 	//initialize channel_stats
 	for(i = 0; i < num_interfaces; i++)
 	{
@@ -381,6 +384,7 @@ void *recv_channel(void *arg)
 	free(recv_req_packet);
 	free(recv_req_header);
 	free(recv_req_data);
+    t = (int32_t*)malloc(sizeof(int32_t));
 
 	pthread_exit((void *)NULL);
 }
\ No newline at end of file
