diff --git a/channels.c b/channels.c
index e6f1587..96c4576 100644
--- a/channels.c
+++ b/channels.c
@@ -1,7 +1,13 @@
 #include "channels.h"
 
+int counter = 0;
+
 void *send_channel(void *arg)
 {
+    pthread_mutex_lock(&counter_l);
+    ++counter;
+    pthread_mutex_unlock(&counter_l);
+    printf("%d", counter);
 	int32_t i;
 	int32_t status;
 	int32_t channel_id = 0;
@@ -232,6 +238,7 @@ void *send_channel(void *arg)
 
 void *recv_channel(void *arg)
 {
+    ++counter;
 	int32_t channel_id;
 	char *recv_req_data;
 	struct mptcp_header *recv_req_header;
diff --git a/global.h b/global.h
index c274eed..06b28f1 100644
--- a/global.h
+++ b/global.h
@@ -58,6 +58,7 @@ byte_stats_t *channel_stats;
 pthread_mutex_t transmission_end_l;
 pthread_mutex_t log_l;
 pthread_mutex_t sys_l;
+pthread_mutex_t counter_l;
 int32_t rwnd;
 int32_t cwnd_total;
 channel_t *channel_wnd;
