diff --git a/channels.c b/channels.c
index e6f1587..bfa43bf 100644
--- a/channels.c
+++ b/channels.c
@@ -1,5 +1,7 @@
 #include "channels.h"
 
+volatile int counter = 0;
+
 void *send_channel(void *arg)
 {
 	int32_t i;
@@ -85,6 +87,7 @@ void *send_channel(void *arg)
 
 	while(transmission_end_sig == 0)
 	{
+	    counter += 1;
 		pthread_mutex_lock(&sys_l);
 		//make sure transmission didn't finish while waiting to obtain the lock
 		if(transmission_end_sig != 0)
@@ -260,6 +263,7 @@ void *recv_channel(void *arg)
 
 	while(transmission_end_sig == 0)
 	{
+	    ++counter;
 		//receive packet
 		errno = 0;
 		(*recv_req_packet->header).ack_num = 0;
